softaware.Authentication.SasToken.AspNetCore/SasTokenAuthenticationHandler.cs

```diff
@@ -0,0 +1,110 @@
+            }
+
+            return AuthenticateResult.Fail("Authentication failed");
+        }
+
+        private async Task<bool> ValidateAsync(HttpRequest request, CancellationToken cancellationToken)
+        {
+            var queryString = request.Query;
+
+            if (!queryString.TryGetValue("sv", out var sasTokenVersionQueryValue) ||
+                !int.TryParse(sasTokenVersionQueryValue, out var sasTokenVersion) ||
+                sasTokenVersion != SasTokenVersion.Version1)
+            {
+                return false;
+            }
+
+            if (!queryString.TryGetValue("st", out var startDateQueryValue) ||
+                !DateTime.TryParse(startDateQueryValue, out var startDate) ||
+                DateTime.UtcNow < startDate)
+            {
+                return false;
+            }
+
+            if (!queryString.TryGetValue("se", out var endDateQueryValue) ||
+                !DateTime.TryParse(endDateQueryValue, out var endDate) ||
+                DateTime.UtcNow > endDate)
+            {
+                return false;
+            }
+
+            if (!queryString.TryGetValue("stt", out var typeQueryValue))
+            {
+                return false;
+            }
+
+            SasTokenType sasTokenType;
+            if (typeQueryValue == "f")
+            {
+                sasTokenType = SasTokenType.ConsiderAllQueryParameters;
+            }
+            else if (typeQueryValue == "p")
+            {
+                sasTokenType = SasTokenType.IgnoreAdditionalQueryParameters;
+            }
+            else
+            {
+                return false;
+            }
+
+            if (!queryString.TryGetValue("sig", out var requestHash))
+            {
+                return false;
+            }
+
+            var generatedHash = await this.sasTokenSignatureGenerator.GenerateAsync(
+                startDate,
+                endDate,
+                sasTokenType,
+                request.Path,
+                GetAdditionalQueryParametersDependingOnType(request, sasTokenType),
+                cancellationToken);
+
+            return requestHash.ToString().Replace(" ", "+") == generatedHash;
+        }
+
+        private static string[] GetAdditionalQueryParametersDependingOnType(HttpRequest request, SasTokenType sasTokenType)
+        {
+            return sasTokenType switch
+            {
+                SasTokenType.ConsiderAllQueryParameters => GetAdditionalQueryParameters(request),
+                SasTokenType.IgnoreAdditionalQueryParameters => [],
+                _ => throw new NotSupportedException($"{nameof(sasTokenType)} {sasTokenType} is not supported."),
+            };
+        }
+
+        private static string[] GetAdditionalQueryParameters(HttpRequest request) =>
+            request.Query
+                .Where(s => s.Key != "sv" && s.Key != "st" && s.Key != "se" && s.Key != "stt" && s.Key != "sig").Select(s => $"{s.Key}={s.Value}")
```

Comment:
Code-Technisch könnte man diese Methode vermutlich auch verbessern, aber die Frage ist halt ob es dafür steht -> die Parameter zB in eine Liste zu geben oder so?!